### ISSUE TITLE
[Feature]: Decide on v4 stereotypes for LocationReferencing

### ISSUE THREAD

**Issue by iancornwellmottmac** [DATEX-II-EU/DatexII#526]
### Feature description

The v4 modelling methodology offers new stereotypes, for entities and for codelists. While we are revising the LocationReferencing namespace for v4, we need to decide whether the stereotypes of any elements need changed. Maybe the answer is none, but we should consider and record our decision here.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by LBlaive** [comment]
Interesting question.
Honestly speaking, after reading the new part 1 (as proposed as prEN 16157-1) I cannot conclude in which case to use the "D2Entity" stereotype or the "D2Class" stereotype.
If I wanted to make a parallel with ISO 19103:2015 and EN ISO 19109:2015, I would associate "D2Entity" with the "FeatureType" stereotype and "D2Class" with "DataType". The distinction as stated in these standards is:
- "DataType" is used for any classifier when "lacking identity". They are often used to typify attributes (which implies they be complex) (if we look at CEN/TS 17268:2018 "TN-ITS" the "Condition" and "Validity" classes are stereotyped as "DataType".
- "FeatureType" is used for "abstraction of real-world phenomena", which means a possibility to define an identity.

With this in mind I would say the "PredefinedLocationReference", "PredefinedLocation", "PredefinedItinerary" and ""PredefinedLocationGroup" could become "D2Entity" and even "D2EntityVersionedIndentifiable". The other classes remaining stereotyped as "D2Class".

But probably my interpretation is incorrect.

About "CodeList", I can think of "Lane", "Carriageway", "InfrastructureDescriptor", "NamedAreaType", "SubdivisionType". 

An open question may be about some TPEGLoc lists if there were a desire to keep consistency with the evolution of these specifications.

**Comment by ingeniumfaber** [comment]
I think Loic proposal is fine

**Comment by JosefKaltwasser** [comment]
About code lists: I assume the proposal refers to Enums. We had agreed that we use CodeList in case where permitted values are governed by an external organisation. These Enums seem to be governed by NAPCORE, and hence I would have expected them to remain to be Enums?

**Comment by iancornwell1** [comment]
I agree with Josef - I can't see a clear benefit in decoupling the evolution of those enumerations from the evolution of the model. For some of them we were discussing their content in other Part 2 issues, and deciding for ourselves what the set of literals should be, which supports their remaining <<D2Enumeration>>.

**Comment by iancornwell1** [comment]
This was discussed in the TC review. WG8 was content with the result, so this issue can be resolved, and in fact since there was no change it can be closed.
